File: string/49.go
func groupAnagrams(strs []string) [][]string {
	var key string
	mapret := make(map[string][]string, 0)
	listret := make([][]string, 0)

	for _, s := range strs {
		key = makeKey(s)
		if _, ok := mapret[key]; !ok {
			mapret[key] = make([]string, 0)
		}
		mapret[key] = append(mapret[key], s)
	}

	for _, v := range mapret {
		listret = append(listret, v)
	}
	return listret    
}

func makeKey(s string) string {
	var temps []string
	var ret string
	for _, c := range s {
		temps = append(temps, string(c))
	}
	sort.Strings(temps)
	for _, c := range temps {
		ret += c
	}
	return ret
}

File: string/49.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        rets = {}

        for s in strs:
            if not rets:
                rets[tuple(sorted(s))] = [s]
                continue
            key = tuple(sorted(s))
            if rets.get(key, None):
                rets[key].append(s)
            else:
                rets[key] = [s]
        return rets.values()
        
